### PR TITLE
admission: move to runtime.DefaultUnstructuredConverter

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of https://github.com/kcp-dev/kcp/pull/508.

### CODE REVIEW

**Comment by sttts** [comment]
e2e still red it seems. Will come back later to this. Not super critical.

File: test/e2e/reconciler/workspace/controller_test.go [hunk 1]
```diff
@@ -186,7 +186,8 @@ func TestWorkspaceController(t *testing.T) {
 			const serverName = "main"
 			f := framework.NewKcpFixture(t,
 				framework.KcpConfig{
-					Name: serverName,
+					Name:         serverName,
+					RunInProcess: true,
```
Comment (marun) [inline test/e2e/reconciler/workspace/controller_test.go:190]: nit: remove

Comment (sttts) [reply test/e2e/reconciler/workspace/controller_test.go]: fixed

**Review by marun — APPROVED** [review]
LGTM

**Pull request merged**